cmd/internal/cli/singularity.go

```diff
@@ -104,7 +105,9 @@ func setSylogMessageLevel(cmd *cobra.Command, args []string) {
 }
 
 func setSylogColor(cmd *cobra.Command, args []string) {
-	if nocolor {
+	// Disable color if nocolor flag, or NOT writing to terminal
+	fd := int(os.Stderr.Fd())
```

Comment:
I don't think we should automatically be trying to do this here. There are too many variables at play for that to make sense in my opinion. I think just providing the flag is good enough

Reply:
This was a suggestion / request of someone else (I can't remember now, @cclerget was it you?) that I added after the fact. It the maintainers want to decide on a consensus I'd be happy to update the functionality.

Reply:
I actually think this is nice to make. If we want color for example when using a pager (e.g. "singularity ... | less") then we need an additional option `--color` or follow git's lead (and other tools' as well) and have `--color=on`, `--color=off`, `--color=auto`, with the last one being the default value. `--color=auto` in git and other tools means "color when the output is a terminal, no color otherwise".

Reply:
Agree! This is similar to what I was bringing up a few days ago :point_up:  https://github.com/sylabs/singularity/pull/2841#issuecomment-471022566

Reply:
Alright, so if this is a standard scheme for coloring output, then I'm happy to oblige here.

Reply:
@mem and @bauerm97 do you want to add the `--color` variable as @mem detailed above, or have two booleans `--nocolor` and `--color` in addition to the automatic detection of where is being written (which is overridden by specifying one of these flags).

Reply:
ping! What is the preference for the argument?

Reply:
@GodloveD I was waiting for feedback here.

Reply:
@vsoch, Imo, I like the variable flag better, just like `ls`, `git` (although git also has a `--no-color` flag).

Reply:
Thanks for the feedback! I had totally forgot about this PR. Let me know what changes would be requested, if any.